doc/CodeReview.md

```diff
@@ -344,6 +344,15 @@ Generally, this means `cfg` directives should be an explicit list of chips or'd
 together. Rarely, if ever, is a `cfg(not ...)` the correct approach for anything
 outside of unit tests.
 
+Rust `cfg` features can also be used to disable code that isn't used in upstream
+Tock but might be used in private or public forks. Rust custom configs can also
+be used, but generally require changes to the lint configuration. See
+https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html#check-cfg-in-lintsrust-table
+for more details on getting the Rust lints to pass.
+
+It would be better to use Cargo feature flags then `cfg`s and avoinding compile
+time configuration at all would be best.
```

Comment:
It's really not about upstream vs not upstream. There are many ways for code to exist in upstream that is used elsewhere without being disabled. It's about using conditional compilation at all, and the use of custom `cfg` predicates in particular.

```suggestion
Custom `cfg` predicates should never be used for conditional compilation unless unavoidable. It is virtually always better to Cargo feature flags, and even better to use non-conditional compilation mechanism (such as generics) to customize functionality at compile-time.
```

Reply:
I was trying to phrase it in a way that covers the [`aon_wkup_timer`](https://github.com/tock/tock/blob/master/chips/lowrisc/src/aon_timer.rs#L146) cfg. It was included with "left for reference/use later", so it seemed like something we should allow in this wording

The new wording doesn't really allow that. Is that really what you want?